MaxText/train.py

```diff
@@ -417,14 +417,39 @@ def create_goodput_recorder(config):
   return None
 
 
-def record_goodput(recorder, config, step=None, job_start=False, job_end=False):
+def record_goodput(
```

Comment:
this API looks pretty funky to me. It seemed okay with only two options, start and end, but with so many options it doesn't make sense to me to have 10 boleans where exactly one is true. I couldn't think of a great alternative due to None checking...  How about

 
```
 def record_goodput(recorder, config, record_func, *args):
     if recorder and config.enable_goodput_recording:
               record_func(*args)
               
# Example call
 record_goodput(recorder,config, recorder.tpu_init_end if recorder else None)
```

Alternatively can avoid None checking with get_attr, which is probably less preferred
```
def record_goodput(recorder, config, record_method, *args):
     if recorder and config.enable_goodput_recording:
               getattr(recorder, record_method)(*args)

 # Example call             
 record_goodput(recorder,config,"tpu_init_end")
```

Reply:
Thanks, done.